crates/ruff_linter/src/rules/pep8_naming/rules/invalid_function_name.rs

```diff
@@ -84,10 +84,64 @@ pub(crate) fn invalid_function_name(
         return;
     }
 
+    let parent_class = semantic
+        .current_statement_parent()
+        .and_then(|parent| parent.as_class_def_stmt());
+
+    // Ignore the visit_* methods of the ast.NodeVisitor and ast.NodeTransformer classes.
+    // Only applies if the Python version is less than 3.12.
+    // If Python is greater than 3.12, typing.override should be used instead.
+    let is_ast_visitor = matches!(
+        checker.target_version(),
+        PythonVersion::PY37
+            | PythonVersion::PY38
+            | PythonVersion::PY39
+            | PythonVersion::PY310
+            | PythonVersion::PY311
+    ) && name.starts_with("visit_")
```

Comment:
nit: I think this could just be < 3.12:


```suggestion
    let is_ast_visitor =  checker.target_version() < PythonVersion::312 
    && name.starts_with("visit_")
```